packages/hub/indexing/branch-update.js

```diff
@@ -99,7 +99,19 @@ class BranchUpdate {
       let newMeta = await updater.updateContent(meta, hints, publicOps, this.uncachedRead);
       await this._saveMeta(updater, newMeta);
     }
-    await this._invalidations();
+    await this.client.invalidations({
+      touched: this._touched,
+      touchCounter: this._touchCounter,
+      schema: await this.schema(),
+      branch: this.branch,
+      read: (type, id) => this.read(type, id),
+      wasInvalidated: async ({ operation, type, id, doc, sourceId, nonce }) => {
+        this.emitEvent(operation, { type, id, doc });
+        if (operation === 'add' && this.isControllingBranch) {
+          await this._maybeUpdateRealms(type, id, doc, sourceId, nonce);
```

Comment:
I think this whole `wasInvalidated` hook should move into the client and be standard for all invalidations.

For example, the realm update thing is needed if somebody is PATCHing a user.

Reply:
so one aspect about moving this into the client that might be tricky is that we also use this to emit indexing events for branch-update, as the invalidation of resources currently triggers indexing events from callers of the pgsearch client which you can see a couple lines up on line 109. would that mean we still need to provide a callback for callers to use?

Reply:
No, we can move the events to also come from pgclient.